src/game/server/swarm/asw_marine.cpp

```diff
@@ -634,6 +635,16 @@ CASW_Marine::CASW_Marine() : m_RecentMeleeHits( 16, 16 )
 
 	m_iPoisonHeal = 0;
 	m_flNextPoisonHeal = -1;
+
+
+	Vector vecCol = Vector(0, 0, 0);
+	Vector vecHSV = Vector(RandomFloat(0.0f, 360.0f), RandomFloat(0.0f, 255.0f), 255.0);
```

Comment:
Instead of random, the default should be the color in g_rgbaStatsReportPlayerColors for the marine resource index

Reply:
Currently, g_rgbaStatsReportPlayerColors is only on the client.

If it's moved into shared then I can use it on the server code.

Reply:
I moved it into a shared file so it can be accessed on server.

I also added code to set lasers to resource index if valid.